### ISSUE TITLE
Darwin support

### ISSUE THREAD

**Issue by roberth** [hercules-ci/arion#84]
- [ ] fix unit tests, add to CI
- [ ] add linux builder setup instructions to README
- [ ] user testing / instructions how to use correctly when `system /= builtins.currentSystem`

**Comment by roberth** [comment]
For podman, 3.2.0 has

> An experimental new set of commands, podman machine, was added to assist in managing virtual machines containing a Podman server. These are intended for easing the use of Podman on OS X by handling the creation of a Linux VM for running Podman.

Alternatively, a module for a VM in `nix-darwin` would be ideal.

**Comment by azazel75** [comment]
Hello, I've an `arion-pkgs.nix` file that pins the `system` to `x86_64-linux` but when I do  an `arion up` on macOS (catalina, x86_64-darwin) it exits with an error:
```
$ arion up
error: a 'x86_64-linux' with features {} is required to build '/nix/store/l86j079x1qkwimjb24601v1s52k5x4lr-builder.pl.drv', but I am a 'x86_64-darwin' with features {benchmark, big-parallel, nixos-test, recursive-nix}
```

is there a way to make this work? anything I can help with ? I'm really debugging the installation of a development environment on macOS because I've a colleague that uses that OS, sadly.. I'm not really an expert on macOS. But If I can help with something just let me know.

**Comment by roberth** [comment]
Yes, you can set up a linux VM to serve as a "remote" builder, using Nix's remote builds via ssh feature.
This question has come up before, but I don't have a good documentation reference yet.
Do you use `nix-darwin`? I figure that could project could provide the means to automate the setup of such a linux builder. (It doesn't yet)

**Comment by Ovyerus** [comment]
I'm relatively new to Nix and I've been using [Colmena](github.com/zhaofengli/colmena/) with `buildOnTarget` to manage a few machines. I'm trying to use Arion on some of them, but I get an error `error: a 'x86_64-linux' with features {} is required to build '/nix/store/ycsvbayyps3vp7pkp0b9l3igxrs35rvg-cabal2nix-arion-compose.drv', but I am a 'aarch64-darwin' with features {benchmark, big-parallel, nixos-test}`.

I'm willing to try and contribute to this, what would need to be done?
